sources/packages/backend/libs/integrations/src/services/sfas/sims-to-sfas.service.ts

```diff
@@ -0,0 +1,161 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { StudentDetail } from "./sims-to-sfas.model";
+import {
+  BC_STUDENT_LOAN_AWARD_CODE,
+  CANADA_STUDENT_LOAN_FULL_TIME_AWARD_CODE,
+} from "@sims/services/constants";
+import {
+  Application,
+  ApplicationStatus,
+  mapFromRawAndEntities,
+  SFASBridgeLog,
+  Student,
+} from "@sims/sims-db";
+import { Brackets, Repository } from "typeorm";
+
+/**
+ * SIMS to SFAS services.
+ */
+@Injectable()
+export class SIMSToSFASService {
+  constructor(
+    @InjectRepository(Student)
+    private readonly studentRepo: Repository<Student>,
+    @InjectRepository(Application)
+    private readonly applicationRepo: Repository<Application>,
+    @InjectRepository(SFASBridgeLog)
+    private readonly sfasBridgeLogRepo: Repository<SFASBridgeLog>,
+  ) {}
+
+  /**
+   * Get the latest bridge file log date.
+   * When there is no log, null will be returned.
+   * @returns latest bridge file log date.
+   */
+  async getLatestBridgeFileLogDate(): Promise<Date | null> {
+    const [latestBridgeFileLog] = await this.sfasBridgeLogRepo.find({
+      select: { id: true, referenceDate: true },
+      order: { referenceDate: "DESC" },
+      take: 1,
+    });
+    return latestBridgeFileLog ? latestBridgeFileLog.referenceDate : null;
+  }
+
+  /**
+   * Log the details of bridge file that was sent to SFAS.
+   * @param referenceDate date when the bridge file data was extracted.
+   * @param fileName bridge file name.
+   */
+  async logBridgeFileDetails(
+    referenceDate: Date,
+    fileName: string,
+  ): Promise<void> {
+    await this.sfasBridgeLogRepo.insert({
+      referenceDate,
+      generatedFileName: fileName,
+    });
+  }
+
+  /**
+   * Get all student ids of students who have one or more updates
+   * since the date provided.
+   * The updates can be one or more of the following:
+   * - Student or User data
+   * - Sin validation data
+   * - Cas supplier data
+   * - Overawards data
+   * @param modifiedSince the date after which the student data was updated.
+   * If not provided, all the students with at least one submitted application will be returned.
+   */
+  async getAllStudentsWithUpdates(modifiedSince?: Date): Promise<number[]> {
```

Comment:
This method would should be called with an optional date once in the SIMS life-time. I would suggest having the date a non-option and letting the consumer decide on a date to be sent to the query.